internal/venice-client-common/src/main/java/com/linkedin/venice/schema/merge/SortBasedCollectionFieldOpHandler.java

```diff
@@ -31,23 +31,29 @@ public SortBasedCollectionFieldOpHandler(AvroCollectionElementComparator element
   public UpdateResultStatus handlePutList(
       final long putTimestamp,
       final int coloID,
-      List<Object> toPutList,
+      Object newFieldValue,
       CollectionRmdTimestamp<Object> collectionFieldRmd,
       GenericRecord currValueRecord,
       String fieldName) {
     if (ignoreIncomingRequest(putTimestamp, coloID, collectionFieldRmd)) {
       return UpdateResultStatus.NOT_UPDATED_AT_ALL;
     }
-    validateFieldSchemaType(currValueRecord, fieldName, Schema.Type.ARRAY, false); // Validate before modifying any
-                                                                                   // state.
+    validateFieldSchemaType(currValueRecord, fieldName, Schema.Type.ARRAY, true);
 
     // Current list will be updated.
     final long currTopLevelTimestamp = collectionFieldRmd.getTopLevelFieldTimestamp();
     collectionFieldRmd.setTopLevelFieldTimestamp(putTimestamp);
     collectionFieldRmd.setTopLevelColoID(coloID);
 
+    List<Object> toPutList;
+    if (newFieldValue == null) {
+      toPutList = Collections.emptyList();
+    } else {
+      toPutList = (List<Object>) newFieldValue;
+    }
+
     if (collectionFieldRmd.isInPutOnlyState()) {
-      currValueRecord.put(fieldName, toPutList);
+      currValueRecord.put(fieldName, newFieldValue);
```

Comment:
I think this is on purpose - because when it is really NULL value then toPutList does not reflect the true value but is the new collection. newFieldValue is the original NULL value.

Reply:
I see.
An empty list is no different from `null` anyway.
Can you add some javadoc to highlight the intention here?

Reply:
Sure, will do.